www/js/config/dynamicConfig.ts

```diff
@@ -224,47 +259,54 @@ function loadNewConfig(newToken, existingVersion = null) {
           configChanged = true;
           return true;
         })
-        .catch((storeError) =>
-          displayError(storeError, i18next.t('config.unable-to-store-config')),
-        );
+        .catch((storeError) => {
+          displayError(storeError, i18next.t('config.unable-to-store-config'));
+          return Promise.reject(storeError);
+        });
     })
     .catch((fetchErr) => {
       displayError(fetchErr, i18next.t('config.unable-download-config'));
+      return Promise.reject(fetchErr);
     });
 }
 
-export function initByUser(urlComponents) {
+// exported wrapper around loadNewConfig that includes error handling
+export function initByUser(urlComponents: { token: string }) {
   const { token } = urlComponents;
   try {
     return loadNewConfig(token).catch((fetchErr) => {
       displayError(fetchErr, i18next.t('config.unable-download-config'));
+      return Promise.reject(fetchErr);
     });
   } catch (error) {
     displayError(error, i18next.t('config.invalid-opcode-format'));
     return Promise.reject(error);
   }
 }
 
-export function resetDataAndRefresh() {
-  // const resetNativePromise = window['cordova'].plugins.BEMUserCache.putRWDocument(CONFIG_PHONE_UI, {});
+/** @description Clears all local and native storage, then triggers a refresh */
+export const resetDataAndRefresh = () =>
   storageClear({ local: true, native: true }).then(() => window.location.reload());
-}
 
-export function getConfig() {
+/**
+ * @returns The app config, either from a cached copy, retrieved from local storage, or retrieved
+ *   from user cache with getDocument()
+ */
+export function getConfig(): Promise<AppConfig> {
   if (storedConfig) return Promise.resolve(storedConfig);
   return storageGet(CONFIG_PHONE_UI_KVSTORE).then((config) => {
     if (config && Object.keys(config).length) {
       logDebug('Got config from KVStore: ' + JSON.stringify(config));
-      storedConfig = config;
-      return config;
+      storedConfig = _backwardsCompatFill(config);
+      return storedConfig;
     }
     logDebug('No config found in KVStore, fetching from native storage');
     return window['cordova'].plugins.BEMUserCache.getDocument(CONFIG_PHONE_UI, false).then(
       (config) => {
         if (config && Object.keys(config).length) {
           logDebug('Got config from native storage: ' + JSON.stringify(config));
-          storedConfig = config;
-          return config;
+          storedConfig = _backwardsCompatFill(config);
+          return storedConfig;
```

Comment:
I know we didn't do this before, but a future enhancement might be to also save the config after backwards filling it. That would ensure that we could eventually remove the backwards compat fill (or turn it into a NOP) instead of having it grow over time.